crates/holochain/src/conductor/manager/mod.rs

```diff
@@ -145,13 +121,99 @@ impl TaskManager {
     }
 }
 
-pub(crate) fn spawn_task_manager(
-    handle: ConductorHandle,
-) -> (mpsc::Sender<ManagedTaskAdd>, TaskManagerRunHandle) {
-    let (send, recv) = mpsc::channel(CHANNEL_SIZE);
-    (send, tokio::spawn(run(handle, recv)))
+pub(crate) fn spawn_task_manager(handle: ConductorHandle) -> (TaskAdder, TaskManagerRunHandle) {
+    let (tx, rx) = mpsc::channel(CHANNEL_SIZE);
+    let adder = TaskAdderImpl {
+        tx,
+        tag: handle.tracing_scope().to_string(),
+    };
+    (TaskAdder(Arc::new(adder)), tokio::spawn(run(handle, rx)))
+}
+
+/// Trait bounds for a managed task
+pub trait ManagedTaskFut: Future<Output = ManagedTaskResult> + Send + 'static {}
+impl<F: Future<Output = ManagedTaskResult> + Send + 'static> ManagedTaskFut for F {}
+
+/// Adds tasks to be managed, instrumented/scoped by a tag for identifiability
+pub struct TaskAdderImpl {
+    tag: String,
+    tx: mpsc::Sender<ManagedTask>,
+}
+
+/// Cloneable signal sender for adding tasks to the task manager.
+/// Contains a "tag" which is used to instrument each task added, so that logs from
+/// different task managers in the same process are distinguishable from each other
+#[derive(Clone, derive_more::Deref)]
+pub struct TaskAdder(Arc<TaskAdderImpl>);
+
+impl TaskAdderImpl {
+    /// Add a task of a certain kind
+    pub async fn add_task(
+        &self,
+        name: &str,
+        kind: TaskKind,
+        fut: impl ManagedTaskFut,
+    ) -> TaskAddResult {
+        let span = tracing::error_span!("conductor", scope = self.tag);
+        let handle = tokio::spawn(fut.instrument(span));
+        self.tx.send(ManagedTask::new(name, kind, handle)).await
+    }
+
+    /// You just want the task in the task manager but don't want
+    /// to react to an error
+    pub async fn ignored_task(&self, name: &str, fut: impl ManagedTaskFut) -> TaskAddResult {
+        self.add_task(name, TaskKind::Ignore, fut).await
+    }
+
+    /// You just want the task in the task manager but don't want
+    /// to react to an error
+    pub async fn persistent_task(&self, name: &str, fut: impl ManagedTaskFut) -> TaskAddResult {
```

Comment:
maybe update the comments on this one--looks like it's a copy/paste from ignored_task